app/src/main/java/com/fastaccess/ui/adapter/viewholder/PullRequestEventViewHolder.kt

```diff
@@ -0,0 +1,368 @@
+                    .append(" ")
+                    .append("unlabeled")
+                    .append(" ")
+                    .append(event.label().name(), CodeSpan(color, ViewHelper.generateTextColor(color), 5.0f))
+                    .append(" ")
+                    .append(ParseDateFormat.getTimeAgo((event.createdAt() as Date)))
+            stateImage.setImageResource(R.drawable.ic_label)
+            avatarLayout.setUrl(it.avatarUrl().toString(), it.login(), false, LinkParserHelper.isEnterprise(it.url().toString()))
+        }
+    }
+
+    private fun unassignedEvent(event: PullRequestTimelineQuery.AsUnassignedEvent) {
+        event.actor()?.let {
+            stateText.text = SpannableBuilder.builder()
+                    .bold(it.login())
+                    .append(" ")
+                    .append("unassigned")
+                    .append(" ")
+                    .append(event.user()?.login())
+                    .append(" ")
+                    .append(ParseDateFormat.getTimeAgo((event.createdAt() as Date)))
+            stateImage.setImageResource(R.drawable.ic_profile)
+            avatarLayout.setUrl(it.avatarUrl().toString(), it.login(), false, LinkParserHelper.isEnterprise(it.url().toString()))
+        }
+    }
+
+    private fun reopenedEvent(event: PullRequestTimelineQuery.AsReopenedEvent) {
+        event.actor()?.let {
+            stateText.text = SpannableBuilder.builder()
+                    .bold(it.login())
+                    .append(" ")
+                    .append("reopened this")
+                    .append(" ")
+                    .append(ParseDateFormat.getTimeAgo((event.createdAt() as Date)))
+            stateImage.setImageResource(R.drawable.ic_issue_opened)
+            avatarLayout.setUrl(it.avatarUrl().toString(), it.login(), false, LinkParserHelper.isEnterprise(it.url().toString()))
+        }
+    }
+
+    private fun renamedEvent(event: PullRequestTimelineQuery.AsRenamedTitleEvent) {
+        event.actor()?.let {
+            stateText.text = SpannableBuilder.builder()
+                    .bold(it.login())
+                    .append(" ")
+                    .append("renamed this from").append(" ").append(event.previousTitle())
+                    .append(" ").append("to").append(" ").bold(event.currentTitle())
+                    .append(" ")
+                    .append(ParseDateFormat.getTimeAgo((event.createdAt() as Date)))
+            stateImage.setImageResource(R.drawable.ic_edit)
+            avatarLayout.setUrl(it.avatarUrl().toString(), it.login(), false, LinkParserHelper.isEnterprise(it.url().toString()))
+        }
+    }
+
+    private fun referenceEvent(event: PullRequestTimelineQuery.AsReferencedEvent) {
+        event.actor()?.let {
+            stateText.text = SpannableBuilder.builder()
+                    .bold(it.login())
+                    .append(" ")
+                    .append(if (event.isCrossRepository) "cross referenced this" else "referenced this")
```

Comment:
This is going to need work.

We will need to know if this is a pull request or not. Leave it for me,